compiler/qsc_circuit/src/builder.rs

```diff
@@ -252,17 +255,24 @@ impl Builder {
             .unwrap_or_default()
     }
 
-    fn finish_circuit(&self, mut circuit: Circuit) -> Circuit {
+    fn finish_circuit(&self, operations: Vec<Operation>) -> Circuit {
+        let mut qubits = vec![];
+
         // add qubit declarations
         for i in 0..self.remapper.num_qubits() {
             let num_measurements = self.num_measurements_for_qubit(WireId(i));
-            circuit.qubits.push(crate::circuit::Qubit {
+            qubits.push(crate::circuit::Qubit {
                 id: i,
-                num_children: num_measurements,
+                num_results: num_measurements,
             });
         }
 
-        circuit
+        let max_q_id = qubits.iter().map(|qubit| qubit.id).max().unwrap_or(0);
```

Comment:
Isn't this just `qubits.len()`? (I think this code used to be more complicated before Stefan's base profile changes, so maybe this is a holdover from that?)

Reply:
Ah yes, this used to be a useful safeguard against an edge case, but I can see that the code has been simplified since then and now it is always just the number of qubits, as you suggest.